client/tracing/src/lib.rs

```diff
@@ -37,12 +38,58 @@ use tracing::{
 	span::{Attributes, Id, Record},
 	subscriber::Subscriber,
 };
-use tracing_subscriber::{CurrentSpan, layer::{Layer, Context}};
+use tracing_subscriber::{
+	fmt::time::ChronoLocal,
+	CurrentSpan,
+	EnvFilter,
+	layer::{self, Layer, Context},
+	fmt as tracing_fmt,
+	Registry,
+};
 
 use sc_telemetry::{telemetry, SUBSTRATE_INFO};
 use sp_tracing::{WASM_NAME_KEY, WASM_TARGET_KEY, WASM_TRACE_IDENTIFIER};
+use tracing_subscriber::reload::Handle;
+use once_cell::sync::OnceCell;
+use tracing_subscriber::filter::Directive;
+
 const ZERO_DURATION: Duration = Duration::from_nanos(0);
 
+// The layered Subscriber as built up in `init_logger()`.
+// Used in the reload `Handle`.
+pub type SCSubscriber<
+	N = tracing_fmt::format::DefaultFields,
+	E = logging::EventFormat<ChronoLocal>,
+	W = fn() -> std::io::Stderr
+> = layer::Layered<tracing_fmt::Layer<Registry, N, E, W>, Registry>;
+
+static FILTER_RELOAD_HANDLE: OnceCell<Handle<EnvFilter, SCSubscriber>> = OnceCell::new();
+
+/// Initialize FILTER_RELOAD_HANDLE, only possible once
+pub fn set_reload_handle(handle: Handle<EnvFilter, SCSubscriber>) {
+	let _ = FILTER_RELOAD_HANDLE.set(handle);
+}
+
+/// Reload the logging filter with the supplied directives
+pub fn reload_filter(directives: String) -> Result<(), String> {
+	let mut env_filter = tracing_subscriber::EnvFilter::default();
+	for directive in parse_directives(directives) {
+		env_filter = env_filter.add_directive(directive);
+	}
+	FILTER_RELOAD_HANDLE.get()
+		.ok_or("No reload handle present".to_string())?
+		.reload(env_filter)
+		.map_err(|e| format!("{}", e))
+}
+
+/// Parse the supplied text directives into `Vec<tracing_subscriber::filter::Directive>`
+pub fn parse_directives(dirs: impl AsRef<str>) -> Vec<Directive> {
+	dirs.as_ref()
+		.split(',')
+		.filter_map(|s| s.parse().ok())
```

Comment:
Maybe we should handle this error? And return it? So that we can inform the caller if something was wrong and couldn't be set? Otherwise people will not know why a directive doesn't work.